Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -173,10 +173,18 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 extension QueryRepresentable where Self: ExecutorRepresentable {
     /// Attempts to delete all entities
     /// in the model's collection.
-    public func delete() throws {
+    public func delete(shouldForceDelete: Bool = true) throws {
```

Comment:
Yeah exactly. `.delete()` should really not actually delete entities no matter where it is called. We also want to ensure that if you have say an array of queries where some entities are soft-deletable and some are not, that looping through that array and calling `.delete()` would soft delete the soft-deletable entities and actually delete normal entities. Making the `.delete()` call dynamic in this way supports that.